HoaryFox/Member/CreateBreps.cs

```diff
@@ -9,96 +9,119 @@ namespace HoaryFox.Member
 {
     public static class CreateBreps
     {
-        public static IEnumerable<Brep> FromEndPoint(StbData stbData, Point3d nodeStart, Point3d nodeEnd, double height, double width, double rotate, ShapeTypes shapeType, FrameType frameType)
+        public static IEnumerable<Brep> FromEndPoint(StbData stbData, ShapeInfo shapeInfo, ShapeTypes shapeType, FrameType frameType)
         {
-            var pointStart = new Point3d[6];
-            var pointEnd = new Point3d[6];
-            List<Brep> brep;
+            Vector3d dVector = shapeInfo.NodeEnd - shapeInfo.NodeStart;
+            var angles = new List<double>
+            {
+                -1 * Math.Atan2(dVector.Y, dVector.X),
+                -1 * Math.Atan2(dVector.Z, dVector.X)
+            };
 
-            double dx = nodeEnd.X - nodeStart.X;
-            double dy = nodeEnd.Y - nodeStart.Y;
-            double dz = nodeEnd.Z - nodeStart.Z;
-            double angleY = -1 * Math.Atan2(dy, dx);
-            double angleZ = -1 * Math.Atan2(dz, dx);
+            List<List<Point3d>> origins = GetOriginPoints(frameType, shapeInfo, angles);
+            List<Brep> breps = GetSecBrep(stbData, shapeType, origins, shapeInfo);
+            List<Brep> brepRot = ApplyRotation(breps, shapeInfo.Rotate, frameType, origins);
+            return brepRot;
+        }
 
-            // 梁は部材天端の中心が起点に対して、柱・ブレースは部材芯が起点なので場合分け
-            switch (frameType)
+        private static List<Brep> ApplyRotation(List<Brep> breps, double rotate, FrameType frameType, List<List<Point3d>> origins)
+        {
+            if (rotate == 0d)
```

Comment:
Codacy found an issue: [Do not check floating point equality with exact values, use a range instead.](https://app.codacy.com/gh/hrntsm/HoaryFox/pullRequest?prid=6416401)